packages/plugins/vcard/vcard.js

```diff
@@ -0,0 +1,54 @@
+
+const xml = require('@xmpp/xml')
+const iqCaller = require('../iq-caller')
+
+const NS_VCARD = 'vcard-temp'
+
+function plugin (entity) {
+  const caller = entity.plugin(iqCaller)
+
+  const _parsevCard =  (el) => {
+    let dict = {}
+    el.children.forEach((c) => {
+      if (c.children && typeof c.children[0] === 'string')
+        dict[c.name] = c.text()
+      else
+        dict[c.name] = _parsevCard(c)
+    })
+    return dict
+  }
+
+  const _buildvCard = (dict, parent) => {
+    let builder = parent ? parent : xml`<vCard xmlns="${NS_VCARD}" version="2.0"/>`
+    for (let [key, val] of Object.entries(dict)) {
+      if (typeof val === 'object')
+        builder.cnode(_buildvCard(val, xml`<${key}/>`)).up()
+      else if (val)
+        builder.c(key).t(val)
+      else
+        builder.c(key).up()
```

Comment:
I think the linter is gonna fail here, make sure that `make test` passes. You'll need prosody 0.10 installed (but no need to configure it) and stopped.